test/Microsoft.Azure.Devices.Edge.Test.Common/Platform.cs

```diff
@@ -1,27 +1,116 @@
 // Copyright (c) Microsoft. All rights reserved.
 namespace Microsoft.Azure.Devices.Edge.Test.Common
 {
-    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
     using System.Runtime.InteropServices;
+    using System.Security.Cryptography.X509Certificates;
     using System.Threading;
     using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Test.Common.Certs;
     using Microsoft.Azure.Devices.Edge.Util;
+    using Serilog;
 
-    public static class Platform
+    public class Platform
     {
-        public static Task<string> CollectDaemonLogsAsync(DateTime testStartTime, string filePrefix, CancellationToken token) => IsWindows()
-            ? Windows.EdgeDaemon.CollectLogsAsync(testStartTime, filePrefix, token)
-            : Linux.EdgeDaemon.CollectLogsAsync(testStartTime, filePrefix, token);
+        public static readonly IPlatform Current = IsWindows() ? new Windows.Platform() as IPlatform : new Linux.Platform();
 
-        // TODO: download installer script from aka.ms if user doesn't pass installerPath in Windows
-        public static IEdgeDaemon CreateEdgeDaemon(Option<string> installerPath) => IsWindows()
-            ? new Windows.EdgeDaemon(installerPath.Expect(() => new ArgumentException()))
-            : new Linux.EdgeDaemon() as IEdgeDaemon;
+        public static async Task<EdgeCertificates> GenerateEdgeCertificatesAsync(
+            string deviceId,
+            string scriptPath,
```

Comment:
why name it as scriptPath? there is no script being called in this method.

Reply:
Now that I've refactored these functions to draw out some common code, it's true: the script to which this path points is not directly invoked from this variable. But that doesn't change the fact that the variable does, in reality, point to the script. And the callers of these functions know it as the "script path" so it didn't think it was worth changing.

Reply:
My latest commit changes scriptPath to point to the directory where the scripts live, not the scripts themselves. As part of that change, I renamed scriptPath to basePath.